### PR TITLE
refactor operator of window func

### PR DESCRIPTION
**Pull request by iamlinjunhong**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/sql/colexec/window/types.go [hunk 1]
```diff
@@ -94,37 +97,91 @@ func (window *Window) Release() {
 }
 
 func (window *Window) Reset(proc *process.Process, pipelineFailed bool, err error) {
-	window.Free(proc, pipelineFailed, err)
+	ctr := &window.ctr
+
+	ctr.resetParam()
```
Comment (ouyuanning) [inline pkg/sql/colexec/window/types.go:102]: 这里可能要重置一下ctr.status

File: pkg/sql/colexec/window/window.go [hunk 2]
```diff
@@ -169,7 +164,7 @@ func (window *Window) Call(proc *process.Process) (vm.CallResult, error) {
 				ctr.status = receive
 			}
 
-			result.Batch = ctr.bat
+			result.Batch = makeResultBatch(ctr.bat, ctr.vec)
```
Comment (ouyuanning) [inline pkg/sql/colexec/window/window.go:167]: 不知道放个 returnBatch 是否合适？纯指针，在reset和free的时候都不动它。仅仅作为返回batch。省得每次都New一个batch？

File: pkg/sql/colexec/window/types.go (hunk 1 above)
Comment (iamlinjunhong) [reply pkg/sql/colexec/window/types.go:102]: resetParam 里有包含

File: pkg/sql/colexec/window/window.go (hunk 2 above)
Comment (iamlinjunhong) [reply pkg/sql/colexec/window/window.go:167]: done

**Pull request merged**